components/tidb_query_datatype/src/codec/batch/lazy_column_vec.rs

```diff
@@ -17,6 +19,20 @@ pub struct LazyBatchColumnVec {
     columns: Vec<LazyBatchColumn>,
 }
 
+impl Index<Range<usize>> for LazyBatchColumnVec {
+    type Output = [LazyBatchColumn];
+
+    fn index(&self, index: Range<usize>) -> &Self::Output {
+        &self.columns[index]
+    }
+}
+
+impl IndexMut<Range<usize>> for LazyBatchColumnVec {
+    fn index_mut(&mut self, index: Range<usize>) -> &mut Self::Output {
+        &mut self.columns[index]
+    }
+}
+
```

Comment:
I guess this does not need to be implemented since `LazyBatchColumnVec` implements `DeRef`?

Reply:
Here is the definition of `IndexMut`, we do need to impl `Deref` first.
```rust
pub trait IndexMut<Idx>: Index<Idx> where
    Idx: ?Sized, {
    fn index_mut(&mut self, index: Idx) -> &mut Self::Output;
}
```

Reply:
Maybe what you want is implementing DerefMut?